packages/react/src/components/ComposedModal/ComposedModal.jsx

```diff
@@ -101,90 +122,96 @@ export const ComposedModalPropTypes = {
  * We also prevent the dialog from closing if you click outside it.
  *
  */
-class ComposedModal extends React.Component {
-  static propTypes = ComposedModalPropTypes;
-
-  static defaultProps = {
-    open: true,
-    error: null,
-    isFetchingData: false,
-    sendingData: null,
-    onClearError: null,
-    type: null,
-    footer: null,
-    isFullScreen: false,
-    isLarge: false,
-    submitFailed: false,
-    onSubmit: null,
-    invalid: false,
-    children: null,
-    header: {},
-    iconDescription: 'Close',
-    passiveModal: false,
-    testId: 'ComposedModal',
-  };
+const ComposedModal = ({
+  header,
+  error,
+  sendingData,
+  children,
+  footer,
+  open,
+  className,
+  type,
+  onClose,
+  isFetchingData,
+  isFullScreen,
+  isLarge,
+  onSubmit,
+  iconDescription,
+  onClearError,
+  submitFailed,
+  invalid,
+  passiveModal,
+  // TODO: remove deprecated testID in v3.
+  testID,
+  testId,
+  ...props
+}) => {
+  useEffect(() => {
+    if (__DEV__ && passiveModal && (footer || onSubmit)) {
+      warning(
+        false,
+        'You have set passiveModal to true, but also passed a footer or onSubmit handler.  Your footer will not be rendered.'
+      );
+    }
+  }, [footer, onSubmit, passiveModal]);
 
-  componentDidUpdate(prevProps) {
-    const { submitFailed, invalid } = this.props;
-    if (invalid && submitFailed !== prevProps.submitFailed) {
+  useEffect(() => {
+    if (invalid && submitFailed) {
       scrollErrorIntoView(true);
     }
-  }
+  }, [invalid, submitFailed]);
 
-  handleClearError = () => {
-    const { onClearError } = this.props;
+  const handleClearError = () => {
     if (onClearError) {
       onClearError();
     }
   };
+  const { height: windowHeight } = useWindowSize();
+  const [maxContentHeight, setMaxContentHeight] = useState();
 
   /** TODO: This is needed to keep the ComposedModal from closing if you click outside it this supports our dialogs on top of dialogs issue */
-  doNotClose = () => false;
+  const doNotClose = () => false;
 
-  render() {
-    const {
-      header,
-      error,
-      sendingData,
-      children,
-      footer,
-      open,
-      className,
-      type,
-      onClose,
-      isFetchingData,
-      isFullScreen,
-      isLarge,
-      onSubmit,
-      iconDescription,
-      onClearError,
-      submitFailed,
-      invalid,
-      passiveModal,
-      // TODO: remove deprecated testID in v3.
-      testID,
-      testId,
-      ...props
-    } = this.props;
-    const { label, title, helpText } = header;
-    // First check for dataErrors as they are worse than form errors
+  const modalWrapperRef = useRef(null);
 
-    if (__DEV__ && passiveModal && (footer || onSubmit)) {
-      warning(
-        false,
-        'You have set passiveModal to true, but also passed a footer or onSubmit handler.  Your footer will not be rendered.'
+  useEffect(() => {
```

Comment:
I wonder if it is better to use `useLayoutEffect` here to prevent the modal from jumping in height on the first render...

Reply:
Yes, I'll make that change.